src/stateless/slider.vue

```diff
@@ -0,0 +1,597 @@
+                    </div>
+                </div>
+
+                <div class="slider-helper-text"></div>
+            </div>
+            
+        </div>
+
+    </div>
+</template>
+
+<script>
+import Input from './input.vue'
+
+export default {
+    components: {
+        inputElement: Input
+    },
+    props: {
+        min: { type: Number },
+        max: { type: Number },
+        limit: { type: Number, required: false },
+        minLabel: { type: String, required: false },
+        maxLabel: { type: String, required: false },
+        label: { type: String },
+        step: { type: Number },
+        value: { type: Number },
+        disabled: { type: Boolean, default: false },
+        disabledText: { type: String, required: false },
+        theme: { type: String, default: 'dark' },
+        size: { type: String, default: 'normal' },
+        alignment: { type: String, default: 'left' },
+        isValid: { type: Function, required: false }
+    },
+    data () {
+        return {
+            domReady: false,
+            index: (this.value - this.min) / this.step,
+            inputValue: null,
+            changedFlag: false,
+            draggingFlag: false,
+            inputFlag: false
+        }
+    },
+    created () {
+        let isStepValid = (this.max - this.min) / this.step % 1 === 0
+        if (!isStepValid)
+            throw new Error('Difference between max and min is not divisible by step')
+
+        let isValueValid = ((this.value - this.min) % this.step).toFixed(0) === '0'
+        if (!isValueValid)
+            throw new Error('Value is not a valid step')
+
+        if (this.min < 0 || 998 < this.min)
+            throw new Error('Min must be between 0 and 998')
+
+        if (this.max > 999 || this.min >= this.max)
+            throw new Error('Max must be between min+1 and 999')
+
+        if (this.value < this.min || this.max < this.value)
+            throw new Error('Value must be between min and max')
+    },
+    mounted () {
+        this.bounds = this.$refs.slider.getBoundingClientRect()
+        this.handle = this.$refs.handle.getBoundingClientRect()
+
+        // threshold for ticks disappearing under labels,  +/- 5 guarantees some padding
+        this.minThreshold = this.$refs.min.getBoundingClientRect().width + 5
+        this.maxThreshold = this.bounds.width - this.$refs.max.getBoundingClientRect().width - 5
+
+        this.domReady = true
+    },
+    computed: {
+        limitValue () {
+            return this.limit || this.max
+        },
+        minLabelValue () {
+            return this.minLabel || this.min.toString()
+        },
+        maxLabelValue () {
+            return this.maxLabel || this.limitValue.toString()
+        },
+        stepsCount () {
+            return (this.limitValue - this.min) / this.step
+        },
+        stepPercentage () {
+            return 1 / this.stepsCount
+        },
+        percentage () {
+            return this.index * this.stepPercentage
```

Comment:
`(this.value - this.min) / (this.max - this.min)` seems cleaner since it calculates this directly from props.